File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/JNullParser.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.{ParseNg, ParseOk, ParseResult, Parser}
import jp.co.teruuu.parser.json.`type`.JNull

object JNullParser extends JParser[JNull] {

  override def parse(input: String): ParseResult[JNull] = {
    val parser = Parser.string("null")
    parser.parse(input) match {
      case result: ParseOk[String] => ParseOk(JNull(), result.next)
      case result: ParseNg[_] => ParseNg(result.message, result.next)
    }
  }
}

File: java/parsers_java8/src/main/java/jp/co/teruuu/parser/json/JNullParser.java
package jp.co.teruuu.parser.json;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.common.Parser;
import jp.co.teruuu.parser.json.type.JNull;
import jp.co.teruuu.parser.json.type.JValue;

public class JNullParser implements JParser<JValue> {
  Parser<String> parser;
  public JNullParser() {
    this.parser = Parser.string("null");
  }

  @Override
  public ParseResult<JValue> parse(String input) {
    String next = input;
    ParseResult result = this.parser.parse(next);
    if(result instanceof ParseResult.Success) {
      next = ((ParseResult.Success<String>) result).next;
      return new ParseResult.Success(new JNull(), next);
    }
    return result;
  }
}
